### PR TITLE
[Rewards 3.0] Add unconnected counter card

### PR DESCRIPTION
**Pull request by zenparsing**
Resolves https://github.com/brave/brave-browser/issues/39723

Screenshot 2024-07-12 at 1 44 00 PM

Screenshot 2024-07-12 at 1 44 24 PM

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/ui/webui/brave_rewards/rewards_page_handler.h [hunk 1]
```diff
@@ -37,31 +41,35 @@ class RewardsPageHandler : public mojom::RewardsPageHandler,
...
                      EnableRewardsCallback callback) override;
   void ResetRewards(ResetRewardsCallback callback) override;
 
-  // RewardsServiceObserver:
-  void OnRewardsInitialized(RewardsService* rewards_service) override;
-  void OnRewardsWalletCreated() override;
-  void OnCompleteReset(bool success) override;
-
  private:
+  class UpdateObserver;
```
Comment (zenparsing) [inline browser/ui/webui/brave_rewards/rewards_page_handler.h:63]: Service observation has been refactored out into a new nested class (defined in the source file).

File: browser/ui/webui/brave_rewards/rewards_page_handler.cc [hunk 2]
```diff
@@ -5,24 +5,92 @@
...
+      ads_service_(brave_ads::AdsServiceFactory::GetForProfile(profile)) {
   CHECK(rewards_service_);
-  rewards_observation_.Observe(rewards_service_);
+  CHECK(ads_service_);
+
+  // Unretained because `update_observer_` is owned by `this`.
+  update_observer_ = std::make_unique<UpdateObserver>(
+      rewards_service_, ads_service_,
+      base::BindRepeating(&RewardsPageHandler::OnUpdate,
+                          base::Unretained(this)));
```
[the remark this answers is not in the document]
Comment (zenparsing) [reply browser/ui/webui/brave_rewards/rewards_page_handler.cc:93]: A code comment above this statement explains the unretained usage.

**Review by mkarolin — APPROVED** [review]
`strings`++

File: browser/ui/webui/brave_rewards/rewards_page_data_source.cc [hunk 3]
```diff
@@ -25,9 +25,12 @@ static constexpr webui::ResourcePath kResources[] = {
     {"favicon.ico", IDR_BRAVE_REWARDS_FAVICON}};
 
 static constexpr webui::LocalizedString kStrings[] = {
-    {"appError", IDS_REWARDS_APP_ERROR_TITLE},
+    {"appErrorTitle", IDS_REWARDS_APP_ERROR_TITLE},
     {"cancelButtonLabel", IDS_REWARDS_PANEL_CANCEL},
     {"closeButtonLabel", IDS_BRAVE_REWARDS_ONBOARDING_CLOSE},
+    {"connectAccountSubtext", IDS_REWARDS_CONNECT_ACCOUNT_TEXT_2},
+    {"connectAccountText", IDS_REWARDS_CONNECT_ACCOUNT_SUBTEXT},
```
Comment (emerick) [inline browser/ui/webui/brave_rewards/rewards_page_data_source.cc:32]: Seems a little confusing that subtext maps to text and vice versa.

Comment (zenparsing) [reply browser/ui/webui/brave_rewards/rewards_page_data_source.cc:32]: Yep - that's not quite right 😄

Comment (zenparsing) [reply browser/ui/webui/brave_rewards/rewards_page_data_source.cc]: Fixed.

**Review by emerick — APPROVED** [review]

**Pull request merged**